examples/IMUKittiExampleGPS.cpp

```diff
@@ -0,0 +1,293 @@
+    noiseModel::Diagonal::shared_ptr sigma_init_v = noiseModel::Isotropic::Sigma(3, 1000.0);
+    noiseModel::Diagonal::shared_ptr sigma_init_b = noiseModel::Isotropic::Sigmas((Vector(6) << Vector3::Constant(0.100), Vector3::Constant(5.00e-05)).finished());
+
+    // Set IMU preintegration parameters
+    Matrix33 measured_acc_cov = Matrix33::Identity(3,3) * pow(AccelerometerSigma,2);
+    Matrix33 measured_omega_cov = Matrix33::Identity(3,3) * pow(GyroscopeSigma,2);
+    Matrix33 integration_error_cov = Matrix33::Identity(3,3) * pow(IntegrationSigma,2); // error committed in integrating position from velocities
+
+    boost::shared_ptr<PreintegratedImuMeasurements::Params> IMU_params = PreintegratedImuMeasurements::Params::MakeSharedU(g);
+    IMU_params->accelerometerCovariance = measured_acc_cov; // acc white noise in continuous
+    IMU_params->integrationCovariance = integration_error_cov; // integration uncertainty continuous
+    IMU_params->gyroscopeCovariance = measured_omega_cov; // gyro white noise in continuous
+    IMU_params->omegaCoriolis = w_coriolis;
+
+    std::shared_ptr<PreintegratedImuMeasurements> currentSummarizedMeasurement = nullptr;
+
+    // Set ISAM2 parameters and create ISAM2 solver object
+    ISAM2Params isamParams;
+    isamParams.factorization = ISAM2Params::CHOLESKY;
+    isamParams.relinearizeSkip = 10;
+
+    ISAM2 isam(isamParams);
+
+    // Create the factor graph and values object that will store new factors and values to add to the incremental graph
+    NonlinearFactorGraph newFactors;
+    Values newValues; // values storing the initial estimates of new nodes in the factor graph
+
+    /// Main loop:
+    /// (1) we read the measurements
+    /// (2) we create the corresponding factors in the graph
+    /// (3) we solve the graph to obtain and optimal estimate of robot trajectory
+    printf("-- Starting main loop: inference is performed at each time step, but we plot trajectory every 10 steps\n");
+    size_t imuMeasurementIndex = 0;
+    for (size_t gpsMeasurementIndex = firstGPSPose; gpsMeasurementIndex < GPS_measurements.size() - 1; gpsMeasurementIndex++) {
+        // At each non=IMU measurement we initialize a new node in the graph
+        auto currentPoseKey = X(gpsMeasurementIndex);
+        auto currentVelKey = V(gpsMeasurementIndex);
+        auto currentBiasKey = B(gpsMeasurementIndex);
+        double t = GPS_measurements[gpsMeasurementIndex].Time;
+
+        if (gpsMeasurementIndex == firstGPSPose) {
+            // Create initial estimate and prior on initial pose, velocity, and biases
+            newValues.insert(currentPoseKey, currentPoseGlobal);
+            newValues.insert(currentVelKey, currentVelocityGlobal);
+            newValues.insert(currentBiasKey, currentBias);
+            newFactors.add(PriorFactor<Pose3>(currentPoseKey, currentPoseGlobal, sigma_init_x));
```

Comment:
use emplace_shared?

Reply:
Addressed in b48bdfdc696328be3fedfe11db18b83d3341e7bc

Reply:
Addressed in b48bdfdc696328be3fedfe11db18b83d3341e7bc